File: backend/app/Models/EmployeeDocument.php
<?php

namespace App\Models;

use App\Enums\DocumentStatus;
use Database\Factories\EmployeeDocumentFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

#[Fillable([
    'employee_id', 'document_type_id', 'title', 'document_number',
    'issue_date', 'expiry_date', 'file_path', 'mime_type', 'file_size',
    'status', 'uploaded_by', 'approved_by', 'approved_at', 'rejection_reason',
])]
class EmployeeDocument extends Model
{
    /** @use HasFactory<EmployeeDocumentFactory> */
    use HasFactory, SoftDeletes;

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class);
    }

    public function documentType(): BelongsTo
    {
        return $this->belongsTo(DocumentType::class);
    }

    public function uploader(): BelongsTo
    {
        return $this->belongsTo(User::class, 'uploaded_by');
    }

    public function approver(): BelongsTo
    {
        return $this->belongsTo(User::class, 'approved_by');
    }

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            // "date:Y-m-d", not plain "date" — see Employee::casts() for why
            // (a bare "date" cast shifts across the day boundary when
            // serialized to JSON under a non-UTC APP_TIMEZONE).
            'issue_date' => 'date:Y-m-d',
            'expiry_date' => 'date:Y-m-d',
            'approved_at' => 'datetime',
            'file_size' => 'integer',
            'status' => DocumentStatus::class,
        ];
    }
}

File: mobile/lib/features/documents/domain/employee_document.dart
class DocumentType {
  DocumentType({required this.id, required this.name, required this.requiresExpiry});

  final int id;
  final String name;
  final bool requiresExpiry;

  factory DocumentType.fromJson(Map<String, dynamic> json) => DocumentType(
        id: json['id'] as int,
        name: json['name'] as String? ?? '',
        requiresExpiry: json['requires_expiry'] as bool? ?? false,
      );
}

class EmployeeDocument {
  EmployeeDocument({
    required this.id,
    this.documentTypeName,
    required this.title,
    this.documentNumber,
    this.issueDate,
    this.expiryDate,
    required this.status,
    this.rejectionReason,
    required this.downloadUrl,
  });

  final int id;
  final String? documentTypeName;
  final String title;
  final String? documentNumber;
  final String? issueDate;
  final String? expiryDate;
  final String status;
  final String? rejectionReason;
  final String downloadUrl;

  factory EmployeeDocument.fromJson(Map<String, dynamic> json) => EmployeeDocument(
        id: json['id'] as int,
        documentTypeName: (json['document_type'] as Map<String, dynamic>?)?['name'] as String?,
        title: json['title'] as String? ?? '',
        documentNumber: json['document_number'] as String?,
        issueDate: json['issue_date'] as String?,
        expiryDate: json['expiry_date'] as String?,
        status: json['status'] as String? ?? 'pending',
        rejectionReason: json['rejection_reason'] as String?,
        downloadUrl: json['download_url'] as String? ?? '',
      );
}
